src/components/DeckDeleteModal.js

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+
+import React from 'react';
+
+import { Modal, ModalBody, ModalFooter, ModalHeader, Button } from 'reactstrap';
+
+import { csrfFetch } from 'utils/CSRF';
+
+class DeckDeleteModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toggle = props.toggle;
+    this.acceptButton = React.createRef();
+    this.confirm = this.confirm.bind(this);
+    this.focusAcceptButton = this.focusAcceptButton.bind(this);
+  }
+
+  focusAcceptButton() {
+    if (this.acceptButton.current) {
+      this.acceptButton.current.focus();
+    }
+  }
+
+  confirm() {
```

Comment:
Couldn't this be async so it doesn't need a then callback?

Reply:
I had copied that code from the BlogPostDelete modal, so I refactored the two modals into a ConfirmDeleteModal component and made both confirm functions use async.